### ISSUE TITLE
[in_app_purchase] Migrate Android Billing Client to 3.0.0

### ISSUE THREAD

**Issue by levent-kantaroglu** [flutter/flutter#75204]
Today I decided to migrate to official _in_app_purchase_ plug-in from _flutter_inapp_purchase_
But when I upload my build to google play console, I faced this warning:

```
Warning
We've detected that this app is using an old version of Google Play Billing. By 1 November 2021, all app updates must use billing library version 3 or newer. Update to billing library 3 before this date. Learn more about billing library 3.
```

When I check the plug-in codes, 
`in_app_purchase 0.3.5+1` uses `implementation 'com.android.billingclient:billing:2.0.3'`
`flutter_inapp_purchase 3.0.1` uses `implementation 'com.android.billingclient:billing:3.0.0'` (I was using this plugin since today)

I wonder why official plug-in do not support Android Billing Client 3.0.0.
I hope it supports as soon as possible.

**Comment by pedromassangocode** [comment]
For reference, this is the dependency that must be updated: https://github.com/flutter/plugins/blob/a87497f1399cf6b074dc98563168569d8f738348/packages/in_app_purchase/android/build.gradle#L38

**Comment by cyanglaz** [comment]
PR is ready. Anyone wants to review it? :)

**Comment by pkaminski2108** [comment]
@cyanglaz I've tried migrate to yours version from PR , but something is not working properly. Is it my fault ? Please take a look on the attachment.
![image](https://user-images.githubusercontent.com/51911742/109858078-aceccb80-7c5b-11eb-81c3-010d01fb25fe.png)

**Comment by cyanglaz** [comment]
@pkaminski2108 I don't see any issue in your code and the same code worked for the example app. Could you provide a complete error message from console?

**Comment by pkaminski2108** [comment]
@cyanglaz take a look
![image](https://user-images.githubusercontent.com/51911742/109872740-1e814580-7c6d-11eb-9af5-80a84af09943.png)

**Comment by cyanglaz** [comment]
This is weird, what is the in_app_purchase version on your pubspec.lock? Could you try flutter pub get to get the latest by running `flutter pub get` again?

**Comment by pkaminski2108** [comment]
@cyanglaz yeah this is weird , version in pubspec.lock is 0.5.0. I've tried `flutter pub get` and nothing has change 
 `  in_app_purchase:
    dependency: "direct main"
    description:
      name: in_app_purchase
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.5.0"`

**Comment by cyanglaz** [comment]
@pkaminski2108 maybe try `flutter clean`? Could you also post your `flutter doctor -v`?

**Comment by pkaminski2108** [comment]
I tried `flutter clean`
`[√] Flutter (Channel stable, 2.0.0, on Microsoft Windows [Version 10.0.19042.804])
    • Flutter version 2.0.0 at C:\Users\pk\fvm\versions\stable
    • Framework revision 60bd88df91 (6 hours ago), 2021-03-03 09:13:17 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\pk\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio      
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] Connected device (3 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 88.0.4324.190

• No issues found!
`

**Comment by cyanglaz** [comment]
Unfortunately, I couldn't tell what went wrong. Was it working if you revert in_app_purchase to a lower version?
What IDE did you use? Was the error coming from your IDE or console/terminal?

**Comment by dvxwang** [comment]
@TahaTesser Can you elaborate on what the fix is here? Should we be downgrading to 0.4.1?
